diff --git a/src/lifecycles/graph.js b/src/lifecycles/graph.js
index 82018af5ffca22010d3ecc5063af053f1ebf8c0c..071a78b173da1db78ca63464434b4feb9ab603ae 100644
--- a/src/lifecycles/graph.js
+++ b/src/lifecycles/graph.js
@@ -360,6 +360,9 @@ const generateGraph = async (compilation) => {
 
       const sourcePlugins = compilation.config.plugins.filter((plugin) => plugin.type === "source");
 
+      compilation.graph = graph;
+      compilation.manifest = { apis };
+
       if (sourcePlugins.length > 0) {
         console.debug("building from external sources...");
         for (const plugin of sourcePlugins) {
@@ -386,9 +389,6 @@ const generateGraph = async (compilation) => {
         }
       }
 
-      compilation.graph = graph;
-      compilation.manifest = { apis };
-
       resolve(compilation);
     } catch (err) {
       reject(err);
