diff --git a/src/data/client.js b/src/data/client.js
index b9c1cccd68d4d0b62e0150bca4d315c891cb3853..19f3b3e0c0857a0c5b391c0f10560cbdeb9ac72d 100644
--- a/src/data/client.js
+++ b/src/data/client.js
@@ -12,9 +12,9 @@ const BASE_PATH = globalThis?.__GWD_BASE_PATH__ ?? "";
 async function getContentAsData(key = "") {
   if (CONTENT_STATE && PRERENDER) {
     // fetch customized query files when a user has opted-in for prerendering with active content
-    await fetch(`${window.location.origin}${BASE_PATH}/data-${key.replace(/\//g, "_")}.json`).then(
-      (resp) => resp.json(),
-    );
+    return await fetch(
+      `${window.location.origin}${BASE_PATH}/data-${key.replace(/\//g, "_")}.json`,
+    ).then((resp) => resp.json());
   } else if (CONTENT_STATE && !PRERENDER) {
     // if user is not prerendering, just fetch the entire graph but apply the same filtering
     const graph = await fetch("/graph.json").then((resp) => resp.json());
